### ISSUE TITLE
Add back `passHref` to the new next/link component

### ISSUE THREAD

**Issue by mahish** [vercel/next.js#41973]
### Describe the feature you'd like to request

Some components might be rendered as an `a` tag if `href` attribute is provided eg. https://developer.microsoft.com/en-us/fluentui#/controls/web/button#row621-15 . This is problem in Next.js v13.

In Next.js v12, this…
```
<Link href={url} passHref>
  <DefaultButton text={text} />
</Link>
```

…rendered in this:
```
<a className="button" href="/">Home</a>
```

In Next.js v13 this is not true and `legacyBehavior` is required to make it work. What are plans with this?

### Describe the solution you'd like

Add back `passHref` prop to `next/link` so this…
```
<Link href={url} passHref>
  <DefaultButton text={text} />
</Link>
```
… is rendereded like this:

```
<a className="button" href="/">Home</a>
```

### Describe alternatives you've considered

The codemod for next/link provided the `legacyBehavior`.

**Comment by grefrit** [comment]
I completely agree. I don't understand what's the point of passing `legacyBehavior` to every link that has `passHref`. It's impossible to get rid of `passHref `anyway, because of UI libraries and so on.

**Comment by seleckis** [comment]
Can confirm that on in v13 next/link produces double `<a>` tag with styled components:
```js
import Link from "next/link";
import styled from "styled-components";

const MyLink = styled.a`
  color: green;
};

const Component = ({ href }) => {
  return (
    <Link passHref href={href}>
      <MyLink>Click Me</MyLink>
    </Link>
  );
}
```
Browser is complaining on 
`Warning: validateDOMNesting(...): <a> cannot appear as a descendant of <a>.`

**Comment by FlorianCassayre** [comment]
This definitely looks like a bug: the `passHref` is supposed to "pass the href", but it is not the case in the newer versions. The so-called "legacy behavior" needs to be brought back.
